src/here-studio.ts

```diff
@@ -0,0 +1,342 @@
+                //Clear cache and delete the file from temp repo
+                await fs.unlinkSync(geoSpaceFileName)
+
+                //Update the modified layer data.
+                updatedLayersData.push(currentLayer)
+            }
+        }
+    }
+
+    //Update the layer data to cloned project - updatedLayersData
+    clonedProjectData.layers = updatedLayersData;
+
+    //Update the token for current project
+    clonedProjectData.rot = currentUsersToken;
+
+    //Create a new project under current user with the settings of published projects
+    let newProjectResponse = await createProject(clonedProjectData, options);
+    let newProjectBody = newProjectResponse.body;
+    let clonedProjectURL = studioBaseURL+"/studio/project/"+newProjectBody.id;
+
+    //Viewer URL - /https://studio.here.com/viewer/?project_id=b1e3a9d4-116b-407b-b99f-17fbdf48d405
+    let viewerURL = studioBaseURL+"/viewer/?project_id="+newProjectBody.id
+    if (newProjectResponse.statusCode == 201) {
+        console.log("\n**** Successfully cloned ******")
+        console.log("Project cloned in your studio account : "+clonedProjectURL)
+        console.log("Viewer URL : "+viewerURL)
+    }
+}
+
+/**
+ * Will create an XYZ Studio project
+ * @param projectData
+ * @param options
+ */
+async function createProject (projectData:any, options:any) {
+    let cType = "application/json"
+    let newProjectResponse = await execute(projectsUrl, "POST", cType, projectData, options.token);
+
+    let newProjectBody = newProjectResponse.body;
+    if (newProjectResponse.statusCode == 201) {
+        console.log("\nProject Created successfully")
+    }
+    else {
+        console.log("\nProject creation failed")
+        return null;
+    }
+    return newProjectResponse
+}
+
+//Will capture URL Query string parameters from URL
+function getParameterByName (name: any, url: any) {
+    if (!url) return null;
+    name = name.replace(/[\[\]]/g, '\\$&');
+    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, ' '));
+}
+
+async function showProject (id : any) {
+    const open = require("open");
+    open(
+        studioBaseURL+"/viewer/?project_id="+id
+        , { wait: false });
+}
+
+async function deleteProject  (id : any, options: any) {
+    console.log("Deleting project : "+id)
+
+    //If project exists send a DELETE request for that projectID
+    const uri = "/project-api/projects/"+id;
+    const cType = "";
+    let response = await execute (uri, "DELETE", cType, "", options.token);
+
+    if (response && response.statusCode === 204) {
+        console.log("Successfully deleted project.")
+    }
+    else {
+        console.log("Unable to delete project having project-id: "+id)
+    }
+}
+
+
+/**
+ * Will fetch all projects
+ * @param options
+ */
+async function findAllProjects (options:any) {
```

Comment:
please rename it to getAllProjects as we are not finding/searching anything based on any parameter

Reply:
Update - Have renamed it to getAllProjects as suggested